src/main/java/spring/intro/dao/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,44 @@
+package spring.intro.dao.impl;
+
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import spring.intro.dao.UserDao;
+import spring.intro.exceptions.DataProcessingException;
+import spring.intro.model.User;
+import java.util.List;
+
+@Repository
+public class UserDaoImpl implements UserDao {
+    @Autowired
+    private SessionFactory sessionFactory;
```

Comment:
What is better: setter auto wiring, constructor auto wiring, or field injection?

Reply:
constructor. changed this part.